src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -65,5 +67,34 @@ public IActionResult GetInstanceInfo()
         {
             return Ok(_instanceManager.GetInstanceInfo());
         }
+
+        /// <summary>
+        /// This endpoint generates a temporary x-ms-site-restricted-token for core tool
+        /// to access KuduLite zipdeploy endpoint in Linux Consumption
+        /// </summary>
+        /// <returns>
+        /// 200 on token generated
+        /// 400 on non-Linux container environment
+        /// 404 on WEBSITE_AUTH_ENCRYPTION_KEY is not set
+        /// </returns>
+        [HttpGet]
+        [Route("admin/getsitetoken")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        public IActionResult GetSiteToken()
+        {
+            if (!_environment.IsLinuxContainerEnvironment())
+            {
+                return BadRequest("Endpoint is only available when running in Linux Container");
+            }
+
+            string websiteEncryptionKey = _environment.GetEnvironmentVariable(EnvironmentSettingNames.WebSiteAuthEncryptionKey);
+            if (string.IsNullOrEmpty(websiteEncryptionKey))
+            {
+                return NotFound("WEBSITE_AUTH_ENCRYPTION_KEY is not set.");
+            }
+
+            string requestHeaderToken = SimpleWebTokenHelper.CreateToken(DateTime.UtcNow.AddHours(2), websiteEncryptionKey.ToKeyBytes());
```

Comment:
The Kudu minted tokens have an expiry of 5 minutes. Why not use the same here?

Reply:
The build process can take up to 2 hours in KuduLite. Since **the core tool polls build status from KuduLite every 3 seconds**, this endpoint needs to be called to generate temporary tokens.

I think making a long expiry token will reduce the call frequency.

Reply:
You'll likely need token refresh logic anyways in core tools - it won't be able to assume the token will be valid long enough for build to complete. If it has to refresh the token every 5 minutes that doesn't seem like a big issue